pkg/controllers/management/rkeworkerupgrader/upgrade.go

```diff
@@ -228,12 +229,25 @@ func (uh *upgradeHandler) upgradeCluster(cluster *v3.Cluster, nodeName string, p
 	uh.clusterLock.Lock(clusterName)
 	defer uh.clusterLock.Unlock(clusterName)
 
-	logrus.Debugf("cluster [%s] upgrading condition: [%v] plan changed: [%v]", clusterName, v3.ClusterConditionUpgraded.IsUnknown(cluster), planChanged)
-	if !v3.ClusterConditionUpgraded.IsUnknown(cluster) || planChanged {
+	upgradeStrategy := cluster.Status.AppliedSpec.RancherKubernetesEngineConfig.UpgradeStrategy
+	toUpdate := !v3.ClusterConditionUpgraded.IsUnknown(cluster) || planChanged || upgradeStrategy == nil
```

Comment:
I don't think we need to check for `upgradeStrategy == nil` while evaluating `toUpdate`, the check for `if upgradeStrategy == nil {` on line 243 should suffice?